modules/urlsigning-service-impl/src/main/java/org/opencastproject/security/urlsigning/provider/impl/WowzaUrlSigningProvider.java

```diff
@@ -46,4 +58,170 @@ public String toString() {
     return "Wowza URL Signing Provider";
   }
 
+  /**
+   * @param policy
+   *             the policy
+   * @return the signed url
+   */
+  @Override
+  public String sign(Policy policy) throws UrlSigningException {
+    if (!accepts(policy.getBaseUrl())) {
+      throw UrlSigningException.urlNotSupported();
+    }
+
+    try {
+      URI uri = new URI(policy.getBaseUrl());
+
+      /*
+        For backward compatibility, but i can not see how this could work.
+        According to the documentation "https://www.wowza.com/docs/how-to-protect-streaming-using-securetoken-in-wowza-streaming-engine"
+        if you using token v1 we need a TEA implementation.
+      */
+      if ("rtmp".equals((uri.getScheme()))) {
```

Comment:
RTMP was removed for Adaptive Streaming anyway a while ago (see #1179). So if this is only used for vod URLs, maybe it can be removed here, too. But I'm not sure.